src/action/mod.rs

```diff
@@ -172,6 +178,14 @@ use crate::error::HasExpectedErrors;
 pub trait Action: Send + Sync + std::fmt::Debug + dyn_clone::DynClone {
     /// A synopsis of the action for tracing purposes
     fn tracing_synopsis(&self) -> String;
+    /// A tracing span suitable for the action
+    ///
+    /// It should be [`tracing::Level::DEBUG`] and contain the same name as the [`typetag::serde`] entry.
+    ///
+    /// It may contain any fields, and will be attached in the [`StatefulAction::try_execute`] and [`StatefulAction::try_revert`] functions.
+    ///
+    /// See [`tracing::Span`] for more details.
+    fn tracing_span(&self) -> Span;
```

Comment:
So I had a minor "whoops" moment where I realized that since `Action::try_execute` and `Action::try_revert` were doing the logging of start/stop of actions this meant that the span from `Action::execute` or `Action::revert` was not always used.

This moves the span to it's own function and properly attaches the span in `try_execute` and `try_revert`.